docs/rules/no-array-index-key.md

```diff
@@ -67,6 +67,14 @@ things.map((thing) => (
   <Hello key={thing.id} />
 ));
 
+things.map((thing, i) => (
+  <Hello key={thing.id + i} />
+));
+
+things.map((thing, i) => (
+  <Hello key={`${thing.id}${i}`} />
+));
```

Comment:
this is an antipattern, though - it shouldn't be allowed.

If the index is present in any way in the key, then all the stuff that's not the index is useless.